hs_composite_resource/models.py

```diff
@@ -57,13 +69,56 @@ def logical_files(self):
         for lf in self.modelinstancelogicalfile_set.all():
             yield lf
 
+    @property
+    def aggregation_types(self):
+        """Gets a list of all aggregation types that currently exist in this resource"""
+        aggr_types = []
+        aggr_type_names = []
+        for lf in self.logical_files:
+            if lf.type_name not in aggr_type_names:
+                aggr_type_names.append(lf.type_name)
+                aggr_type = lf.get_aggregation_display_name().split(":")[0]
+                aggr_types.append(aggr_type)
+        return aggr_types
+
+    def get_logical_files(self, logical_file_class_name):
+        """Get a list of logical files (aggregations) for a specified logical file class name."""
+
+        class_name_to_query_mappings = dict()
+        class_name_to_query_mappings["GenericLogicalFile"] = self.genericlogicalfile_set.all()
+        class_name_to_query_mappings["NetCDFLogicalFile"] = self.netcdflogicalfile_set.all()
+        class_name_to_query_mappings["GeoRasterLogicalFile"] = self.georasterlogicalfile_set.all()
+        class_name_to_query_mappings["GeoFeatureLogicalFile"] = self.geofeaturelogicalfile_set.all()
+        class_name_to_query_mappings["FileSetLogicalFile"] = self.filesetlogicalfile_set.all()
+        class_name_to_query_mappings["ModelProgramLogicalFile"] = self.modelprogramlogicalfile_set.all()
+        class_name_to_query_mappings["ModelInstanceLogicalFile"] = self.modelinstancelogicalfile_set.all()
+        class_name_to_query_mappings["TimeSeriesLogicalFile"] = self.timeserieslogicalfile_set.all()
+        class_name_to_query_mappings["RefTimeseriesLogicalFile"] = self.reftimeserieslogicalfile_set.all()
```

Comment:
Don't use ```dict()``` to initialize a dictionary but rather use `{}` due to performance reasons. See http://katrin-affolter.ch/Python/Python_dictionaries for detailed explanation. Even better, I think you can just directly define the dict as follows: 
```
class_name_to_query_mappings = {
    "GenericLogicalFile": self.genericlogicalfile_set.all(),
   ....
}
```

Reply:
Fixed